libs/dst/ProcPathStasCollection.py

```diff
@@ -0,0 +1,98 @@
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+# GNU Affero General Public License for more details.
+# You should have received a copy of the GNU Affero General Public License
+# along with this program. If not, see <https://www.gnu.org/licenses/>.
+#
+# For any questions about this software or licensing,
+# please email opensource@seagate.com or cortx-questions@seagate.com.
+#
+
+"""Procpath memory stats collector """
+
+import logging
+from multiprocessing import Process
+from commons.helpers.pods_helper import LogicalNode
+
+proc_cmd = "pid=$(echo $(pgrep m0d; pgrep radosgw; pgrep hax) | sed -z 's/ /,/g'); procpath " \
+           "record -i 45 -d s3server-$(date +%d%m_%H%M%S).sqlite -p $pid"
+
+pid_watch_list = ['m0d', 'radosgw', 'hax']
+
+# check and set pytest logging level as Globals.LOG_LEVEL
+LOGGER = logging.getLogger(__name__)
+
+
+class EnableProcPathStatsCollection:
+
+    def __init__(self, cmn_cfg):
+        """Initialize connection to Nodes or Pods."""
+        self.cmn_cfg = cmn_cfg
+        self.nodes = cmn_cfg["nodes"]
+        self.worker_node_list = list()
+        for node in self.nodes:
+            if node["node_type"].lower() == "worker":
+                node_obj = LogicalNode(hostname=node["hostname"],
+                                       username=node["username"],
+                                       password=node["password"])
+                self.worker_node_list.append(node_obj)
+        self.stat_collection = []
+
+    @staticmethod
+    def stat_collection_cmd(worker: LogicalNode, cmd: str):
+        """command for Procpath collection"""
+        LOGGER.debug("executing procpath cmd for stat collection")
+        worker.execute_cmd(cmd=cmd)
+
+    @staticmethod
+    def collect_pids(worker, watch_list):
+        """function to collect pids """
+        for proc in watch_list:
+            res = worker.execute_cmd("echo $(pgrep {})".format(proc))
+            logging.info(f'{proc} PIDs {res}')
+            with open("{}".format(worker.hostname), 'a') as fp:
+                fp.write(f'{proc} PIDs {res}')
+
+    def setup_requirement(self):
+        """Install required modules on worker nodes for procpath collection"""
+        LOGGER.debug("installing required modules")
+        for worker_node in self.worker_node_list:
+            worker_node.execute_cmd("pip install Procpath apsw-wheels")
+
+    def start_collection(self):
+        """trigger command for collection stats"""
+        LOGGER.debug("starting stat collection")
+        for worker in self.worker_node_list:
+            self.collect_pids(worker, pid_watch_list)
+        for worker in self.worker_node_list:
+            self.stat_collection.append(Process(target=self.stat_collection_cmd, args=(worker,
+                                                                                       proc_cmd)))
+        for proc in self.stat_collection:
+            proc.start()
```

Comment:
do we want to handle start and stop from test instead in case we want it at particular time
cc: @priyankaborawake

Reply:
we are handling start and stop in different function